Compilation of druid-shell fails on windows

Hi,
I wanted to give druid a try, but I'm unable to compile druid for windows.
This is the error I get:
```
Compiling druid-shell v0.4.0
error[E0308]: mismatched types
 --> C:\Users\Zwolin\.cargo\registry\src\github.com-1ecc6299db9ec823\druid-shell-0.4.0\src\platform\windows\window.rs:1007:48
     |
1007 |         SetWindowLongPtrW(hwnd, GWLP_USERDATA, wndproc_ptr as LONG_PTR);
     |                                                ^^^^^^^^^^^^^^^^^^^^^^^ expected i32, found isize
```
And the toolchain I use:
```
stable-i686-pc-windows-msvc
rustc 1.40.0 (73528e339 2019-12-16)
```

--- a/druid-shell/src/platform/windows/window.rs
+++ b/druid-shell/src/platform/windows/window.rs
@@ -26,7 +26,6 @@ use std::sync::{Arc, Mutex};
 
 use log::{debug, error, warn};
 use winapi::ctypes::{c_int, c_void};
-use winapi::shared::basetsd::*;
 use winapi::shared::dxgi::*;
 use winapi::shared::dxgi1_2::*;
 use winapi::shared::dxgiformat::*;
@@ -995,6 +994,11 @@ unsafe fn create_dcomp_state(
     }
 }
 
+#[cfg(target_arch = "x86_64")]
+type WindowLongPtr = winapi::shared::basetsd::LONG_PTR;
+#[cfg(target_arch = "x86")]
+type WindowLongPtr = LONG;
+
 pub(crate) unsafe extern "system" fn win_proc_dispatch(
     hwnd: HWND,
     msg: UINT,
@@ -1004,7 +1008,7 @@ pub(crate) unsafe extern "system" fn win_proc_dispatch(
     if msg == WM_CREATE {
         let create_struct = &*(lparam as *const CREATESTRUCTW);
         let wndproc_ptr = create_struct.lpCreateParams;
-        SetWindowLongPtrW(hwnd, GWLP_USERDATA, wndproc_ptr as LONG_PTR);
+        SetWindowLongPtrW(hwnd, GWLP_USERDATA, wndproc_ptr as WindowLongPtr);
     }
     let window_ptr = GetWindowLongPtrW(hwnd, GWLP_USERDATA) as *const WindowState;
     let result = {

